mod/calculated_metrics/lib/formula/calculator/input_item/value_store.rb

```diff
@@ -0,0 +1,30 @@
+module Formula
+  class Calculator
+    class InputItem
+      # Used to hold onto the values that an input item finds
+      class ValueStore
+        attr_reader :years, :values
+        def initialize
+          @years = ::Set.new
+          @companies = ::Set.new
+          @values = Hash.new_nested(Hash, Hash)
+        end
+
```

Comment:
can we add a `@return` comment here?

If I understand correctly, it returns:
`[Hash] { company_id => { year => value }` if company is nil and year is nil or missing
`{ year => value }` if company is valid and year is nil or missing,
`value` if both are present.

I think it would also try to look up company_id based on a year if company is nil and year is valid, but that's an edge case :)